Automattic/wp-calypso
# Themes: Validate search filters

![screen shot 2016-08-02 at 16 28 38](https://cloud.githubusercontent.com/assets/7767559/17334455/40cca368-58ce-11e6-8bdb-3f55977fbd27.png)

Adds filter validation using a static table of theme taxonomies.
* As you type into the search box, only complete and valid _taxonomy:term_ filters (from the `taxonomies` table) will be queried and added to the URL
* When entering a URL, filters will be reconstructed and added to the search box

`taxonomies` object created using this script: [source](https://github.com/seear/theme-taxonomies/blob/master/theme_tax.php). At some point we may want to dynamically fetch it.

Test live: https://calypso.live/?branch=add/themes-filter-table

**Todo**
- [x] ~~validate filter url and redirect in the controller~~ Tracking in #7225
- [x] validate filter input

**To Test**
Input
* go to /design and enter filter queries in the search box in the form _taxonomy:term_
* Only complete and valid filters should be added to the URL and passed to the API call
* You can enter `localStorage.setItem('debug', 'calypso:themes:actions' )` in the console to see the `filter` queries in the API call

URLs
* Enter a URL such as `http://calypso.localhost:3000/design/filter/blue,business,post-slider`
* The filters should be reconstructed correctly in the search box
* Invalid slugs in the URL will not be added to the search box, but currently will be sent in the API call (this will be addressed in another PR)
